File: audit-public-pages.mjs
/**
 * Playwright audit script for Purple Glow Social 2.0 public pages
 */
import { chromium } from '@playwright/test';
import { writeFileSync, mkdirSync } from 'fs';
import { join } from 'path';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const PROJECT_ROOT = process.cwd();
const SCREENSHOTS_DIR = join(PROJECT_ROOT, 'docs', 'audit-screenshots');
const BASE_URL = 'http://localhost:3000';

// Ensure screenshots directory exists
mkdirSync(SCREENSHOTS_DIR, { recursive: true });

const PAGES = [
    { url: '/', slug: 'home' },
    { url: '/login', slug: 'login' },
    { url: '/signup', slug: 'signup' },
    { url: '/privacy', slug: 'privacy' },
    { url: '/terms', slug: 'terms' },
];

async function auditPage(page, url, slug) {
    console.log(`\n${'='.repeat(60)}`);
    console.log(`Auditing: ${url}`);
    console.log(`${'='.repeat(60)}`);

    const consoleErrors = [];
    const consoleWarnings = [];
    const networkErrors = [];
    const visualObservations = [];
    const accessibilityObservations = [];

    // Set up console listener
    page.on('console', (msg) => {
        const msgType = msg.type();
        const msgText = msg.text();
        if (msgType === 'error') {
            consoleErrors.push(msgText);
            console.log(`  [CONSOLE ERROR] ${msgText}`);
        } else if (msgType === 'warning') {
            consoleWarnings.push(msgText);
            console.log(`  [CONSOLE WARNING] ${msgText}`);
        }
    });

    // Set up network listener
    page.on('response', (response) => {
        const status = response.status();
        if (status >= 400) {
            const errorMsg = `${response.request().method()} ${response.url()} - ${status}`;
            networkErrors.push(errorMsg);
            console.log(`  [NETWORK ERROR] ${errorMsg}`);
        }
    });

    // Navigate to page
    console.log(`  Navigating to ${BASE_URL}${url}...`);
    try {
        await page.goto(`${BASE_URL}${url}`, { waitUntil: 'networkidle', timeout: 30000 });
        console.log(`  ✓ Page loaded successfully`);
    } catch (e) {
        console.log(`  ✗ Failed to load page: ${e.message}`);
        return {
            url,
            slug,
            error: e.message,
            consoleErrors,
            consoleWarnings,
            networkErrors,
            visualObservations: [`Failed to load: ${e.message}`],
            accessibilityObservations: []
        };
    }

    // Wait a bit for any async operations
    await page.waitForTimeout(1000);

    // Take screenshot
    const screenshotPath = join(SCREENSHOTS_DIR, `${slug}-public.png`);
    console.log(`  Taking screenshot: ${screenshotPath}`);
    await page.screenshot({ path: screenshotPath, fullPage: true });
    console.log(`  ✓ Screenshot saved`);

    // Visual observations
    console.log(`  Analyzing visual layout...`);

    // Check for broken images
    const brokenImages = await page.evaluate(() => {
        const images = Array.from(document.querySelectorAll('img'));
        return images
            .filter((img) => !img.complete || img.naturalHeight === 0)
            .map((img) => img.src || img.alt || 'unnamed image');
    });
    if (brokenImages.length > 0) {
        brokenImages.forEach((img) => {
            visualObservations.push(`Broken image: ${img}`);
            console.log(`  ⚠ Broken image: ${img}`);
        });
    }

    // Check for horizontal overflow
    const hasOverflow = await page.evaluate(() => {
        return document.body.scrollWidth > document.body.clientWidth;
    });
    if (hasOverflow) {
        visualObservations.push('Horizontal overflow detected');
        console.log(`  ⚠ Horizontal overflow detected`);
    }

    // Check for elements outside viewport
    const overflowingElements = await page.evaluate(() => {
        const elements = Array.from(document.querySelectorAll('*'));
        const bodyWidth = document.body.clientWidth;
        return elements
            .filter((el) => {
                const rect = el.getBoundingClientRect();
                return rect.right > bodyWidth + 10;
            })
            .slice(0, 5)
            .map((el) => {
                const tag = el.tagName.toLowerCase();
                const id = el.id ? `#${el.id}` : '';
                const cls = el.className ? `.${el.className.split(' ')[0]}` : '';
                return `${tag}${id}${cls}`;
            });
    });
    if (overflowingElements.length > 0) {
        overflowingElements.forEach((el) => {
            visualObservations.push(`Element overflows viewport: ${el}`);
            console.log(`  ⚠ Element overflows: ${el}`);
        });
    }

    // Accessibility observations
    console.log(`  Analyzing accessibility...`);

    // Check for images without alt text
    const imagesNoAlt = await page.evaluate(() => {
        const images = Array.from(document.querySelectorAll('img'));
        return images
            .filter((img) => !img.alt || img.alt.trim() === '')
            .map((img) => img.src || img.outerHTML.substring(0, 50))
            .slice(0, 5);
    });
    if (imagesNoAlt.length > 0) {
        imagesNoAlt.forEach((img) => {
            accessibilityObservations.push(`Image missing alt text: ${img}`);
            console.log(`  ⚠ Image missing alt: ${img}`);
        });
    }

    // Check for buttons without accessible text
    const unlabeledButtons = await page.evaluate(() => {
        const buttons = Array.from(document.querySelectorAll('button, [role="button"]'));
        return buttons
            .filter((btn) => {
                const text = btn.textContent.trim();
                const label = btn.getAttribute('aria-label');
                const labelledby = btn.getAttribute('aria-labelledby');
                return !text && !label && !labelledby;
            })
            .map((btn) => btn.outerHTML.substring(0, 100))
            .slice(0, 5);
    });
    if (unlabeledButtons.length > 0) {
        unlabeledButtons.forEach((btn) => {
            accessibilityObservations.push(`Unlabeled button: ${btn}`);
            console.log(`  ⚠ Unlabeled button: ${btn}`);
        });
    }

    // Check for inputs without labels
    const unlabeledInputs = await page.evaluate(() => {
        const inputs = Array.from(document.querySelectorAll('input:not([type="hidden"])'));
        return inputs
            .filter((input) => {
                const id = input.id;
                const label = input.getAttribute('aria-label');
                const labelledby = input.getAttribute('aria-labelledby');
                const hasLabel = id && document.querySelector(`label[for="${id}"]`);
                const placeholder = input.placeholder;
                return !label && !labelledby && !hasLabel && !placeholder;
            })
            .map((input) => {
                const type = input.type;
                const name = input.name;
                return `${type} input (name: ${name})`;
            })
            .slice(0, 5);
    });
    if (unlabeledInputs.length > 0) {
        unlabeledInputs.forEach((inp) => {
            accessibilityObservations.push(`Unlabeled input: ${inp}`);
            console.log(`  ⚠ Unlabeled input: ${inp}`);
        });
    }

    // Page-specific checks
    if (url === '/') {
        console.log(`  Running homepage-specific checks...`);

        // Check navigation links
        const navLinks = await page.evaluate(() => {
            const nav = document.querySelector('nav');
            if (!nav) return [];
            const links = Array.from(nav.querySelectorAll('a'));
            return links.map((a) => ({
                text: a.textContent.trim(),
                href: a.href,
                hasHref: !!a.href
            }));
        });
        if (navLinks.length > 0) {
            console.log(`  ✓ Found ${navLinks.length} navigation links`);
            navLinks.forEach((link) => {
                if (!link.hasHref) {
                    visualObservations.push(`Navigation link '${link.text}' missing href`);
                }
            });
        } else {
            visualObservations.push('No navigation links found');
        }

        // Check pricing section
        const pricingInfo = await page.evaluate(() => {
            const pricingSection = document.body.textContent || '';
            const hasZAR = pricingSection.includes('R') || pricingSection.includes('ZAR');
            const hasPricing = pricingSection.toLowerCase().includes('price') || 
                              pricingSection.toLowerCase().includes('plan');
            return { hasZAR, hasPricing };
        });
        if (pricingInfo.hasPricing) {
            if (pricingInfo.hasZAR) {
                console.log(`  ✓ Pricing section found with ZAR currency`);
            } else {
                visualObservations.push('Pricing section missing ZAR currency (R)');
            }
        } else {
            visualObservations.push('Pricing section not found');
        }

        // Check footer
        const footerCheck = await page.evaluate(() => {
            const footer = document.querySelector('footer');
            if (!footer) return { exists: false };
            const rect = footer.getBoundingClientRect();
            const bodyWidth = document.body.clientWidth;
            return {
                exists: true,
                overflows: rect.right > bodyWidth + 10
            };
        });
        if (footerCheck.exists) {
            if (footerCheck.overflows) {
                visualObservations.push('Footer has horizontal overflow');
                console.log(`  ⚠ Footer overflows`);
            } else {
                console.log(`  ✓ Footer renders without overflow`);
            }
        } else {
            visualObservations.push('Footer not found');
        }
    }

    // Summary
    console.log(`  Console errors: ${consoleErrors.length}`);
    console.log(`  Console warnings: ${consoleWarnings.length}`);
    console.log(`  Network errors: ${networkErrors.length}`);
    console.log(`  Visual issues: ${visualObservations.length}`);
    console.log(`  Accessibility issues: ${accessibilityObservations.length}`);

    return {
        url,
        slug,
        consoleErrors,
        consoleWarnings,
        networkErrors,
        visualObservations,
        accessibilityObservations
    };
}

function generateReport(results) {
    // Count critical issues
    const totalConsoleErrors = results.reduce((sum, r) => sum + r.consoleErrors.length, 0);
    const totalNetworkErrors = results.reduce((sum, r) => sum + r.networkErrors.length, 0);
    const totalVisual = results.reduce((sum, r) => sum + r.visualObservations.length, 0);
    const totalA11y = results.reduce((sum, r) => sum + r.accessibilityObservations.length, 0);

    // Determine overall status
    let status;
    if (totalConsoleErrors > 0 || totalNetworkErrors > 0) {
        status = '⚠️ Issues found requiring attention';
    } else if (totalVisual > 5 || totalA11y > 5) {
        status = '⚠️ Minor issues found';
    } else {
        status = '✅ All pages audited successfully';
    }

    // Build report
    let report = `# Public Pages Audit Report

**Audit Date:** ${new Date().toISOString()}
**Base URL:** ${BASE_URL}
**Pages Audited:** ${results.length}

## Summary

${status}

**Totals:**
- Console errors: ${totalConsoleErrors}
- Network errors: ${totalNetworkErrors}
- Visual observations: ${totalVisual}
- Accessibility observations: ${totalA11y}

## Page Results

`;

    // Add each page
    results.forEach(result => {
        const { url, slug, error, consoleErrors, consoleWarnings, networkErrors, 
                visualObservations, accessibilityObservations } = result;

        // Handle errors
        if (error) {
            report += `### ${url} (${slug.charAt(0).toUpperCase() + slug.slice(1)})
- **Screenshot:** docs/audit-screenshots/${slug}-public.png
- **Status:** ❌ Failed to load
- **Error:** ${error}

`;
            return;
        }

        report += `### ${url} (${slug.charAt(0).toUpperCase() + slug.slice(1)})
- **Screenshot:** docs/audit-screenshots/${slug}-public.png
- **Console errors:** ${consoleErrors.length > 0 ? consoleErrors.length : 'none'}
`;
        if (consoleErrors.length > 0) {
            consoleErrors.slice(0, 5).forEach(err => {
                report += `  - ${err}\n`;
            });
            if (consoleErrors.length > 5) {
                report += `  - ... and ${consoleErrors.length - 5} more\n`;
            }
        }

        report += `- **Console warnings:** ${consoleWarnings.length > 0 ? consoleWarnings.length : 'none'}\n`;
        if (consoleWarnings.length > 0) {
            consoleWarnings.slice(0, 3).forEach(warn => {
                report += `  - ${warn}\n`;
            });
            if (consoleWarnings.length > 3) {
                report += `  - ... and ${consoleWarnings.length - 3} more\n`;
            }
        }

        report += `- **Network errors:** ${networkErrors.length > 0 ? networkErrors.length : 'none'}\n`;
        if (networkErrors.length > 0) {
            networkErrors.forEach(err => {
                report += `  - ${err}\n`;
            });
        }

        report += `- **Visual observations:** ${visualObservations.length > 0 ? visualObservations.length : 'none'}\n`;
        if (visualObservations.length > 0) {
            visualObservations.forEach(obs => {
                report += `  - ${obs}\n`;
            });
        }

        report += `- **Accessibility observations:** ${accessibilityObservations.length > 0 ? accessibilityObservations.length : 'none'}\n`;
        if (accessibilityObservations.length > 0) {
            accessibilityObservations.forEach(obs => {
                report += `  - ${obs}\n`;
            });
        }

        report += '\n';
    });

    // Critical issues section
    report += '## Critical Issues Requiring Immediate Attention\n\n';

    const criticalIssues = [];
    results.forEach(result => {
        const { url, error, consoleErrors, networkErrors } = result;
        
        if (error) {
            criticalIssues.push(`**${url}**: Failed to load - ${error}`);
        }

        // Console errors are critical
        consoleErrors.forEach(err => {
            if (err.includes('Failed to fetch') || err.includes('TypeError') || err.includes('ReferenceError')) {
                criticalIssues.push(`**${url}**: ${err}`);
            }
        });

        // Network 500 errors are critical
        networkErrors.forEach(err => {
            if (err.includes('500') || err.includes('503')) {
                criticalIssues.push(`**${url}**: ${err}`);
            }
        });
    });

    if (criticalIssues.length > 0) {
        criticalIssues.slice(0, 10).forEach(issue => {
            report += `- ${issue}\n`;
        });
        if (criticalIssues.length > 10) {
            report += `- ... and ${criticalIssues.length - 10} more\n`;
        }
    } else {
        report += 'No critical issues found.\n';
    }

    report += '\n---\n\n*This report was generated automatically by Playwright audit script.*\n';

    return report;
}

async function run() {
    console.log('='.repeat(60));
    console.log('Purple Glow Social 2.0 - Public Pages Audit');
    console.log('='.repeat(60));
    console.log(`Base URL: ${BASE_URL}`);
    console.log(`Screenshots: ${SCREENSHOTS_DIR}`);
    console.log(`Pages to audit: ${PAGES.length}`);
    console.log();

    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({ viewport: { width: 1920, height: 1080 } });
    const page = await context.newPage();

    const results = [];
    for (const pageInfo of PAGES) {
        const result = await auditPage(page, pageInfo.url, pageInfo.slug);
        results.push(result);
    }

    await browser.close();
    console.log('\n' + '='.repeat(60));
    console.log('Audit complete!');
    console.log('='.repeat(60));

    // Generate report
    console.log('\nGenerating report...');
    const reportContent = generateReport(results);
    const reportPath = join(SCREENSHOTS_DIR, 'report-public.md');
    writeFileSync(reportPath, reportContent, 'utf-8');
    console.log(`✓ Report saved to: ${reportPath}`);

    return reportPath;
}

run()
    .then(reportPath => {
        console.log(`\n✅ Audit complete! Report: ${reportPath}`);
        process.exit(0);
    })
    .catch(err => {
        console.error('❌ Audit failed:', err);
        process.exit(1);
    });

File: audit_public_pages.py
"""
Playwright audit script for Purple Glow Social 2.0 public pages
"""
from playwright.sync_api import sync_playwright
import json
from pathlib import Path
from typing import List, Dict, Any

# Project root
PROJECT_ROOT = Path(r"C:\scratchpad\purple-glow-social-2.0")
SCREENSHOTS_DIR = PROJECT_ROOT / "docs" / "audit-screenshots"
BASE_URL = "http://localhost:3000"

# Pages to audit
PAGES = [
    {"url": "/", "slug": "home"},
    {"url": "/login", "slug": "login"},
    {"url": "/signup", "slug": "signup"},
    {"url": "/privacy", "slug": "privacy"},
    {"url": "/terms", "slug": "terms"},
]

class PageAudit:
    def __init__(self):
        self.results = []
        
    def audit_page(self, page, url: str, slug: str) -> Dict[str, Any]:
        """Audit a single page"""
        print(f"\n{'='*60}")
        print(f"Auditing: {url}")
        print(f"{'='*60}")
        
        # Storage for issues
        console_errors = []
        console_warnings = []
        network_errors = []
        visual_observations = []
        accessibility_observations = []
        
        # Set up console listener
        def handle_console(msg):
            msg_type = msg.type
            msg_text = msg.text
            if msg_type == 'error':
                console_errors.append(msg_text)
                print(f"  [CONSOLE ERROR] {msg_text}")
            elif msg_type == 'warning':
                console_warnings.append(msg_text)
                print(f"  [CONSOLE WARNING] {msg_text}")
        
        page.on('console', handle_console)
        
        # Set up network listener
        def handle_response(response):
            status = response.status
            if status >= 400:
                error_msg = f"{response.request.method} {response.url} - {status}"
                network_errors.append(error_msg)
                print(f"  [NETWORK ERROR] {error_msg}")
        
        page.on('response', handle_response)
        
        # Navigate to page
        print(f"  Navigating to {BASE_URL}{url}...")
        try:
            page.goto(f"{BASE_URL}{url}", wait_until='networkidle', timeout=30000)
            print(f"  ✓ Page loaded successfully")
        except Exception as e:
            print(f"  ✗ Failed to load page: {e}")
            return {
                'url': url,
                'slug': slug,
                'error': str(e),
                'console_errors': console_errors,
                'console_warnings': console_warnings,
                'network_errors': network_errors,
                'visual_observations': [f"Failed to load: {e}"],
                'accessibility_observations': []
            }
        
        # Wait a bit for any async operations
        page.wait_for_timeout(1000)
        
        # Take screenshot
        screenshot_path = SCREENSHOTS_DIR / f"{slug}-public.png"
        print(f"  Taking screenshot: {screenshot_path}")
        page.screenshot(path=str(screenshot_path), full_page=True)
        print(f"  ✓ Screenshot saved")
        
        # Visual observations
        print(f"  Analyzing visual layout...")
        
        # Check for broken images
        broken_images = page.evaluate("""
            () => {
                const images = Array.from(document.querySelectorAll('img'));
                return images
                    .filter(img => !img.complete || img.naturalHeight === 0)
                    .map(img => img.src || img.alt || 'unnamed image');
            }
        """)
        if broken_images:
            for img in broken_images:
                visual_observations.append(f"Broken image: {img}")
                print(f"  ⚠ Broken image: {img}")
        
        # Check for horizontal overflow
        has_overflow = page.evaluate("""
            () => {
                return document.body.scrollWidth > document.body.clientWidth;
            }
        """)
        if has_overflow:
            visual_observations.append("Horizontal overflow detected")
            print(f"  ⚠ Horizontal overflow detected")
        
        # Check for elements outside viewport
        overflowing_elements = page.evaluate("""
            () => {
                const elements = Array.from(document.querySelectorAll('*'));
                const bodyWidth = document.body.clientWidth;
                return elements
                    .filter(el => {
                        const rect = el.getBoundingClientRect();
                        return rect.right > bodyWidth + 10; // 10px tolerance
                    })
                    .slice(0, 5) // Limit to 5
                    .map(el => {
                        const tag = el.tagName.toLowerCase();
                        const id = el.id ? `#${el.id}` : '';
                        const cls = el.className ? `.${el.className.split(' ')[0]}` : '';
                        return `${tag}${id}${cls}`;
                    });
            }
        """)
        if overflowing_elements:
            for el in overflowing_elements:
                visual_observations.append(f"Element overflows viewport: {el}")
                print(f"  ⚠ Element overflows: {el}")
        
        # Accessibility observations
        print(f"  Analyzing accessibility...")
        
        # Check for images without alt text
        images_no_alt = page.evaluate("""
            () => {
                const images = Array.from(document.querySelectorAll('img'));
                return images
                    .filter(img => !img.alt || img.alt.trim() === '')
                    .map(img => img.src || img.outerHTML.substring(0, 50))
                    .slice(0, 5);
            }
        """)
        if images_no_alt:
            for img in images_no_alt:
                accessibility_observations.append(f"Image missing alt text: {img}")
                print(f"  ⚠ Image missing alt: {img}")
        
        # Check for buttons without accessible text
        unlabeled_buttons = page.evaluate("""
            () => {
                const buttons = Array.from(document.querySelectorAll('button, [role="button"]'));
                return buttons
                    .filter(btn => {
                        const text = btn.textContent.trim();
                        const label = btn.getAttribute('aria-label');
                        const labelledby = btn.getAttribute('aria-labelledby');
                        return !text && !label && !labelledby;
                    })
                    .map(btn => btn.outerHTML.substring(0, 100))
                    .slice(0, 5);
            }
        """)
        if unlabeled_buttons:
            for btn in unlabeled_buttons:
                accessibility_observations.append(f"Unlabeled button: {btn}")
                print(f"  ⚠ Unlabeled button: {btn}")
        
        # Check for inputs without labels
        unlabeled_inputs = page.evaluate("""
            () => {
                const inputs = Array.from(document.querySelectorAll('input:not([type="hidden"])'));
                return inputs
                    .filter(input => {
                        const id = input.id;
                        const label = input.getAttribute('aria-label');
                        const labelledby = input.getAttribute('aria-labelledby');
                        const hasLabel = id && document.querySelector(`label[for="${id}"]`);
                        const placeholder = input.placeholder;
                        return !label && !labelledby && !hasLabel && !placeholder;
                    })
                    .map(input => {
                        const type = input.type;
                        const name = input.name;
                        return `${type} input (name: ${name})`;
                    })
                    .slice(0, 5);
            }
        """)
        if unlabeled_inputs:
            for inp in unlabeled_inputs:
                accessibility_observations.append(f"Unlabeled input: {inp}")
                print(f"  ⚠ Unlabeled input: {inp}")
        
        # Page-specific checks
        if url == "/":
            print(f"  Running homepage-specific checks...")
            
            # Check navigation links
            nav_links = page.evaluate("""
                () => {
                    const nav = document.querySelector('nav');
                    if (!nav) return [];
                    const links = Array.from(nav.querySelectorAll('a'));
                    return links.map(a => ({
                        text: a.textContent.trim(),
                        href: a.href,
                        hasHref: !!a.href
                    }));
                }
            """)
            if nav_links:
                print(f"  ✓ Found {len(nav_links)} navigation links")
                for link in nav_links:
                    if not link['hasHref']:
                        visual_observations.append(f"Navigation link '{link['text']}' missing href")
            else:
                visual_observations.append("No navigation links found")
            
            # Check pricing section
            pricing_info = page.evaluate("""
                () => {
                    const pricingSection = document.body.textContent;
                    const hasZAR = pricingSection.includes('R') || pricingSection.includes('ZAR');
                    const hasPricing = pricingSection.toLowerCase().includes('price') || 
                                      pricingSection.toLowerCase().includes('plan');
                    return { hasZAR, hasPricing };
                }
            """)
            if pricing_info['hasPricing']:
                if pricing_info['hasZAR']:
                    print(f"  ✓ Pricing section found with ZAR currency")
                else:
                    visual_observations.append("Pricing section missing ZAR currency (R)")
            else:
                visual_observations.append("Pricing section not found")
            
            # Check footer
            footer_check = page.evaluate("""
                () => {
                    const footer = document.querySelector('footer');
                    if (!footer) return { exists: false };
                    const rect = footer.getBoundingClientRect();
                    const bodyWidth = document.body.clientWidth;
                    return {
                        exists: true,
                        overflows: rect.right > bodyWidth + 10
                    };
                }
            """)
            if footer_check['exists']:
                if footer_check['overflows']:
                    visual_observations.append("Footer has horizontal overflow")
                    print(f"  ⚠ Footer overflows")
                else:
                    print(f"  ✓ Footer renders without overflow")
            else:
                visual_observations.append("Footer not found")
        
        # Summary
        print(f"  Console errors: {len(console_errors)}")
        print(f"  Console warnings: {len(console_warnings)}")
        print(f"  Network errors: {len(network_errors)}")
        print(f"  Visual issues: {len(visual_observations)}")
        print(f"  Accessibility issues: {len(accessibility_observations)}")
        
        return {
            'url': url,
            'slug': slug,
            'console_errors': console_errors,
            'console_warnings': console_warnings,
            'network_errors': network_errors,
            'visual_observations': visual_observations,
            'accessibility_observations': accessibility_observations
        }
    
    def generate_report(self, results: List[Dict[str, Any]]) -> str:
        """Generate markdown report"""
        
        # Count critical issues
        total_console_errors = sum(len(r.get('console_errors', [])) for r in results)
        total_network_errors = sum(len(r.get('network_errors', [])) for r in results)
        total_visual = sum(len(r.get('visual_observations', [])) for r in results)
        total_a11y = sum(len(r.get('accessibility_observations', [])) for r in results)
        
        # Determine overall status
        if total_console_errors > 0 or total_network_errors > 0:
            status = "⚠️ Issues found requiring attention"
        elif total_visual > 5 or total_a11y > 5:
            status = "⚠️ Minor issues found"
        else:
            status = "✅ All pages audited successfully"
        
        # Build report
        report = f"""# Public Pages Audit Report

**Audit Date:** {Path(__file__).stat().st_mtime}
**Base URL:** {BASE_URL}
**Pages Audited:** {len(results)}

## Summary

{status}

**Totals:**
- Console errors: {total_console_errors}
- Network errors: {total_network_errors}
- Visual observations: {total_visual}
- Accessibility observations: {total_a11y}

## Page Results

"""
        
        # Add each page
        for result in results:
            url = result['url']
            slug = result['slug']
            
            # Handle errors
            if 'error' in result:
                report += f"""### {url} ({slug.title()})
- **Screenshot:** docs/audit-screenshots/{slug}-public.png
- **Status:** ❌ Failed to load
- **Error:** {result['error']}

"""
                continue
            
            console_errors = result.get('console_errors', [])
            console_warnings = result.get('console_warnings', [])
            network_errors = result.get('network_errors', [])
            visual_obs = result.get('visual_observations', [])
            a11y_obs = result.get('accessibility_observations', [])
            
            report += f"""### {url} ({slug.title()})
- **Screenshot:** docs/audit-screenshots/{slug}-public.png
- **Console errors:** {len(console_errors) if console_errors else "none"}
"""
            if console_errors:
                for err in console_errors[:5]:  # Limit to 5
                    report += f"  - {err}\n"
                if len(console_errors) > 5:
                    report += f"  - ... and {len(console_errors) - 5} more\n"
            
            report += f"- **Console warnings:** {len(console_warnings) if console_warnings else "none"}\n"
            if console_warnings:
                for warn in console_warnings[:3]:  # Limit to 3
                    report += f"  - {warn}\n"
                if len(console_warnings) > 3:
                    report += f"  - ... and {len(console_warnings) - 3} more\n"
            
            report += f"- **Network errors:** {len(network_errors) if network_errors else "none"}\n"
            if network_errors:
                for err in network_errors:
                    report += f"  - {err}\n"
            
            report += f"- **Visual observations:** {len(visual_obs) if visual_obs else "none"}\n"
            if visual_obs:
                for obs in visual_obs:
                    report += f"  - {obs}\n"
            
            report += f"- **Accessibility observations:** {len(a11y_obs) if a11y_obs else "none"}\n"
            if a11y_obs:
                for obs in a11y_obs:
                    report += f"  - {obs}\n"
            
            report += "\n"
        
        # Critical issues section
        report += "## Critical Issues Requiring Immediate Attention\n\n"
        
        critical_issues = []
        for result in results:
            url = result['url']
            if 'error' in result:
                critical_issues.append(f"**{url}**: Failed to load - {result['error']}")
            
            # Console errors are critical
            for err in result.get('console_errors', []):
                if 'Failed to fetch' in err or 'TypeError' in err or 'ReferenceError' in err:
                    critical_issues.append(f"**{url}**: {err}")
            
            # Network 500 errors are critical
            for err in result.get('network_errors', []):
                if '500' in err or '503' in err:
                    critical_issues.append(f"**{url}**: {err}")
        
        if critical_issues:
            for issue in critical_issues[:10]:  # Limit to 10
                report += f"- {issue}\n"
            if len(critical_issues) > 10:
                report += f"- ... and {len(critical_issues) - 10} more\n"
        else:
            report += "No critical issues found.\n"
        
        report += "\n---\n\n*This report was generated automatically by Playwright audit script.*\n"
        
        return report
    
    def run(self):
        """Run the full audit"""
        print("="*60)
        print("Purple Glow Social 2.0 - Public Pages Audit")
        print("="*60)
        print(f"Base URL: {BASE_URL}")
        print(f"Screenshots: {SCREENSHOTS_DIR}")
        print(f"Pages to audit: {len(PAGES)}")
        print()
        
        with sync_playwright() as p:
            print("Launching browser...")
            browser = p.chromium.launch(headless=True)
            context = browser.new_context(viewport={'width': 1920, 'height': 1080})
            page = context.new_page()
            
            results = []
            for page_info in PAGES:
                result = self.audit_page(page, page_info['url'], page_info['slug'])
                results.append(result)
            
            browser.close()
            print("\n" + "="*60)
            print("Audit complete!")
            print("="*60)
        
        # Generate report
        print("\nGenerating report...")
        report_content = self.generate_report(results)
        report_path = SCREENSHOTS_DIR / "report-public.md"
        report_path.write_text(report_content, encoding='utf-8')
        print(f"✓ Report saved to: {report_path}")
        
        return report_path

if __name__ == "__main__":
    auditor = PageAudit()
    report_path = auditor.run()
    print(f"\n✅ Audit complete! Report: {report_path}")
